src/lib/event/DY.Lib.Event.Typed.fst

```diff
@@ -41,35 +41,40 @@ let mk_event_instance #a #format tag = {
 type event_predicate (a:Type0) {|event a|} =
   trace -> principal -> a -> prop
 
-let split_event_pred_func: split_predicate_input_values = {
+let split_event_pred_func: split_function_input_values = {
   tagged_data_t = trace & principal & string & bytes;
+
+  tag_set_t = string;
   tag_t = string;
-  encoded_tag_t = string;
+  is_disjoint = default_disjoint;
+  tag_belong_to = (fun dtag tag -> dtag = tag);
+  cant_belong_to_disjoint_sets = (fun dtag tag1 tag2 -> ());
+
   raw_data_t = trace & principal & bytes;
+  output_t = prop;
+
+  default_global_fun = (fun tr prin tag content -> False);
```

Comment:
i think this should come after `global_fun` to match the order in the definition of the `split_function_input_values`, and also to know the type of this function before defining it.

Reply:
Good catch, I changed that!